libbeat/_meta/config/output-elasticsearch.reference.yml.tmpl

```diff
@@ -81,6 +81,11 @@ output.elasticsearch:
   # Elasticsearch after a network error. The default is 60s.
   #backoff.max: 60s
 
+  # The maximum amount of time an idle connection will remain idle
+  # before closing itself.  Zero means no limit. The format is a Go
+  # language duration (example 60s is 60 seconds). The default is 0.
+  #idle_connection_timeout: 0
```

Comment:
Is the default being 0 actually the behavior from before. There are a lot of layers to follow in elastic-agent-libs, but I see when we create an HTTP round tripper we clone the default transport:

https://github.com/elastic/elastic-agent-libs/blob/9ec47704eb509f0944507fa11eb48a580711bbba/transport/httpcommon/httpcommon.go#L254-L259

```go
func (settings *HTTPTransportSettings) httpRoundTripper(
	tls *tlscommon.TLSConfig,
	dialer, tlsDialer transport.Dialer,
	opts ...TransportOption,
) *http.Transport {
	t := http.DefaultTransport.(*http.Transport).Clone()
```

The default transport uses a 90 second timeout by default:

https://cs.opensource.google/go/go/+/refs/tags/go1.21.3:src/net/http/transport.go;l=38-54

```go
// DefaultTransport is the default implementation of Transport and is
// used by DefaultClient. It establishes network connections as needed
// and caches them for reuse by subsequent calls. It uses HTTP proxies
// as directed by the environment variables HTTP_PROXY, HTTPS_PROXY
// and NO_PROXY (or the lowercase versions thereof).
var DefaultTransport RoundTripper = &Transport{
	Proxy: ProxyFromEnvironment,
	DialContext: defaultTransportDialContext(&net.Dialer{
		Timeout:   30 * time.Second,
		KeepAlive: 30 * time.Second,
	}),
	ForceAttemptHTTP2:     true,
	MaxIdleConns:          100,
	IdleConnTimeout:       90 * time.Second,
	TLSHandshakeTimeout:   10 * time.Second,
	ExpectContinueTimeout: 1 * time.Second,
}
```

Reply:
To be clear `0` might be what this configuration was set to, but I am not sure the result `IdleConnTimeout` was `0`, I think it was `90s` because `IdleConnTimeout` is only written if it is *not* zero which means it uses the underlying Go default:

https://github.com/elastic/elastic-agent-libs/blob/9ec47704eb509f0944507fa11eb48a580711bbba/transport/httpcommon/httpcommon.go#L309-L313

```go
func (opts WithKeepaliveSettings) applyTransport(_ *HTTPTransportSettings, t *http.Transport) {
	t.DisableKeepAlives = opts.Disable
	if opts.IdleConnTimeout != 0 {
		t.IdleConnTimeout = opts.IdleConnTimeout
	}
```

Reply:
We were both wrong.  The esleg client sets it to 60s by default.

https://github.com/elastic/beats/blob/42f2f9426043d5ee6fe76ec490e74c9680fdd1b0/libbeat/esleg/eslegclient/connection.go#L105-L107

This also matches testing I did with Wireshark where I observed the default at 60s.

Reply:
![](https://media.giphy.com/media/RSOUOj8H9A3Xq/giphy.gif)